typescript/lib/util/configUtils.ts

```diff
@@ -0,0 +1,52 @@
+import { JSONObject } from "../../common";
+import { InferenceSettings, ModelMetadata } from "../../types";
+import { AIConfigRuntime } from "../config";
+import _ from "lodash";
+
+/**
+ *  Extract inference settings with overrides based on inference settings.
+ *
+ *    This function takes the inference settings and a model ID and returns a subset
+ *    of inference settings that have been overridden by model-specific settings. It
+ *    compares the provided settings with global settings, and returns only those that
+ *    differ or have no corresponding global setting.
+ * @param configRuntime The AIConfigRuntime that the prompt belongs to.
+ * @param inferenceSettings The model settings from the input data.
+ * @param modelId The model ID of the prompt.
+ * @returns The model settings from the input data.
+ */
+export function extractOverrideSettings(
+  configRuntime: AIConfigRuntime,
+  inferenceSettings: InferenceSettings,
+  modelId: string
```

Comment:
`modelName` since that's what we call it everywhere

Reply:
updated